velox/functions/sparksql/aggregates/CentralMomentsAggregate.cpp

```diff
@@ -64,28 +96,49 @@ exec::AggregateRegistrationResult registerCentralMoments(
           core::AggregationNode::Step step,
           const std::vector<TypePtr>& argTypes,
           const TypePtr& resultType,
-          const core::QueryConfig& /*config*/)
-          -> std::unique_ptr<exec::Aggregate> {
+          const core::QueryConfig& config) -> std::unique_ptr<exec::Aggregate> {
         VELOX_CHECK_EQ(argTypes.size(), 1, "{} takes only one argument", name);
         const auto& inputType = argTypes[0];
-        if (exec::isRawInput(step)) {
-          if (inputType->kind() == TypeKind::DOUBLE) {
-            return std::make_unique<
-                CentralMomentsAggregatesBase<double, TResultAccessor>>(
-                resultType);
+        if (config.sparkLegacyStatisticalAggregate()) {
+          if (exec::isRawInput(step)) {
+            if (inputType->kind() == TypeKind::DOUBLE) {
```

Comment:
Could be replaced with a user check for whether 'inputType->kind()' is 'TypeKind::DOUBLE'.